src/main/java/seedu/recipe/ui/AddRecipeForm.java

```diff
@@ -0,0 +1,286 @@
+     * @return commands The StringBuilder instance containing the command string, to be passed back to saveRecipe().
+     */
+    private StringBuilder handleAddRecipeEvent(Map<String, String> inputValues) {
+        commands.append("add");
+        // Check if the name has been added and append the name prefix and value.
+        if (inputValues.containsKey("name")) {
+            commands.append(" n/");
+            commands.append(inputValues.get("name"));
+        }
+
+        // Check if the duration has been added and append the duration prefix and value.
+        if (inputValues.containsKey("duration")) {
+            commands.append(" d/");
+            commands.append(inputValues.get("duration"));
+        }
+
+        // Check if the portion has been added and append the duration prefix and value.
+        if (inputValues.containsKey("portion")) {
+            commands.append(" p/");
+            commands.append(inputValues.get("portion"));
+        }
+
+        // Check if the ingredients have been added and append the ingredients prefix and value.
+        if (inputValues.containsKey("ingredients")) {
+            String[] ingredients = inputValues.get("ingredients").split(", ");
+            for (String ingredient : ingredients) {
+                commands.append(" i/");
+                commands.append(ingredient);
+            }
+        }
+
+        // Check if the steps have been added and append the steps prefix and value.
+        if (inputValues.containsKey("steps")) {
+            String[] steps = inputValues.get("steps").split(", ");
+            for (String step : steps) {
+                commands.append(" s/");
+                commands.append(step);
+            }
+        }
+
+        // Check if the tags have been added and append the tags prefix and value.
+        if (inputValues.containsKey("tags")) {
+            String[] tags = inputValues.get("tags").split(", ");
+            for (String tag : tags) {
+                commands.append(" t/");
+                commands.append(tag);
+            }
+        }
+        return commands;
+    }
+
+    /**
+     * Returns the next TextField below the current TextField within the same parent VBox, if any.
+     *
+     * @param currentTextField The current TextField.
+     * @return The next TextField below the current TextField or null if there's no TextField below it.
+     */
+    private TextField getNextTextField(TextField currentTextField) {
+        VBox parentBox = (VBox) currentTextField.getParent();
+        int currentIndex = parentBox.getChildren().indexOf(currentTextField);
+        int lastIndex = parentBox.getChildren().size() - 1;
+        if (currentIndex < lastIndex) {
+            Node nextNode = parentBox.getChildren().get(currentIndex + 1);
+            if (nextNode instanceof TextField) {
+                return (TextField) nextNode;
+            }
+        }
+        return null;
```

Comment:
not sure how the UI components work but is it safe to return `null` here? will this cause any `NullPointerException`s?